### PR TITLE
Clean error msg in case of boxplot

### LINKED ISSUE #478: Boxplot error in non-integer column
Dataset: https://www.kaggle.com/competitions/house-prices-advanced-regression-techniques

```
%sqlplot boxplot --table train.csv --column LotFrontage
```

This results in error:
```
ProgrammingError: (duckdb.BinderException) Binder Error: No function matches the given name and argument types 'avg(VARCHAR)'. You might need to add explicit type casts.
	Candidate functions:
	avg(DECIMAL) -> DECIMAL
	avg(SMALLINT) -> DOUBLE
	avg(INTEGER) -> DOUBLE
	avg(BIGINT) -> DOUBLE
	avg(HUGEINT) -> DOUBLE
	avg(DOUBLE) -> DOUBLE

LINE 1: ...RDER BY "LotFrontage") AS percentiles, AVG("LotFrontage") AS mean, COUNT(*) AS...
                                                  ^
[SQL: SELECT PERCENTILE_DISC(LIST_VALUE(0.25, 0.50, 0.75)) WITHIN GROUP (ORDER BY "LotFrontage") AS percentiles, AVG("LotFrontage") AS mean, COUNT(*) AS N FROM "train.csv"]
(Background on this error at: https://sqlalche.me/e/20/f405)
```

Probably because the column is non-integer.

We can 

1. Display more meaningful error message like `Ensure column is a number type.` and add Slack link
2. Convert these type of columns to float and handle missing values/ null values.

### PR DESCRIPTION
**Pull request by mehtamohit013**
## Describe your changes
Clean error msg in case of non-numeric column for boxplot
## Issue number

Closes #478 

## Checklist before requesting a review

- [X] Performed a self-review of my code
- [X] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [X] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [X] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--829.org.readthedocs.build/en/829/

### CODE REVIEW

**Comment by mehtamohit013** [comment]
@edublancas 
Ready for review and merge

File: src/sql/plot.py [hunk 1]
```diff
@@ -233,18 +233,21 @@ def boxplot(payload, table, column, *, orient="v", with_=None, conn=None, ax=Non
     vert = orient == "v"
 
     set_ticklabels = ax.set_xticklabels if vert else ax.set_yticklabels
+    set_ticks = ax.set_xticks if vert else ax.set_yticks
```
Comment (edublancas) [inline src/sql/plot.py:236]: is this some unrelated bug that you fixed? doesn't seem related to the PR

File: src/sql/stats.py [hunk 2]
```diff
@@ -107,10 +107,21 @@ def _summary_stats_parallel(conn, table, column, with_=None):
...
+        msg_numeric_type = "You might need to add explicit type casts"
         print(e)
-        raise exceptions.RuntimeError(
-            f"\nEnsure that percentile_disc function is available on {driver}."
-        )
+
+        if (
+            type(e.orig).__name__ == "UndefinedFunction"
+            or type(e.orig).__name__ == "BinderException"
+        ) and msg_numeric_type in e.orig.args[0]:
```
Comment (edublancas) [inline src/sql/stats.py:116]: comment on when each one is raised, I'm guessing one is for postgres and another one for duckdb?

File: src/sql/stats.py [hunk 3]
```diff
@@ -107,10 +107,21 @@ def _summary_stats_parallel(conn, table, column, with_=None):
...
+            raise exceptions.RuntimeError(
+                "Boxplot currently only supports numeric types. "
+                "Ensure that the column is of numeric type."
+            )
+        else:
+            raise exceptions.RuntimeError(
+                f"\nEnsure that percentile_disc function is available on {driver}."
+            )
```
Comment (edublancas) [inline src/sql/stats.py:124]: when raising an exception inside another exception use `raise Exception(...) from e` to chain them

**Review by edublancas — CHANGES_REQUESTED** [review]

**Pull request closed without merging**